src/lib/operators/multi_predicate_join.hpp

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include <memory>
+#include <vector>
+
+#include "all_type_variant.hpp"
+#include "expression/binary_predicate_expression.hpp"
+#include "expression/evaluation/expression_evaluator.hpp"
+#include "expression/evaluation/expression_result.hpp"
+#include "expression/value_expression.hpp"
+#include "storage/table.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+inline AllTypeVariant _get_value(const Table& table, const RowID row_id, const ColumnID& column_id) {
```

Comment:
`RowID` should be a reference.
```suggestion
inline AllTypeVariant _get_value(const Table& table, const RowID& row_id, const ColumnID& column_id) {
```

Reply:
According to https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#f16-for-in-parameters-pass-cheaply-copied-types-by-value-and-others-by-reference-to-const it actually might be better to pass row_id and column_id by value.

Reply:
I'm not sure for `RowID`, since it is not a simple integer but a struct: https://github.com/hyrise/hyrise/blob/master/src/lib/types.hpp#L103-L132

Reply:
I don't know. RowID is still only 8 bytes.

Reply:
Indeed, it doesn't seem to make a difference:
http://quick-bench.com/AALp1c4NMXnCqyVTmjayN8xeHFE
![aalp1c4nmxncqyvtmjayn8xehfe](https://user-images.githubusercontent.com/2145092/52786497-44722800-305b-11e9-8bc1-9f660c1429a8.png)

Reply:
I way just going to try that :-D